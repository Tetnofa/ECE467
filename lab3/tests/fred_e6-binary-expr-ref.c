int main() {
  bool b1;
  bool b2;
  bool b3;
  bool b4;
  bool b5;
  int i1;
  int i2;
  int i3;
  int i4;
  int i5;
  int i6;

  b1 &&b2;
  b1 || b2;
  i1 + i2;
  i1 - i2;
  i1 *i2;
  i1 / i2;
  i1 == i2;
  i1 != i2;
  b1 == b2;
  b1 != b2;
  i1 < i2;
  i1 <= i2;
  i1 > i2;
  i1 >= i2;
  (b1 && b2) || b3;
  (i1 + i2) - i3;
  (i1 * i2) / (i3 + i4);
  (b1 || b2) && (b3 || b4);
  ((i1 * i2) - i3) < i4;
  (i1 + (i2 * i3)) == i4;
  (b1 && (b2 || b3)) == (b4 && b5);
  i1 + i2 *i3 - i4 / i5;
  (i1 + i2 * i3) <= (i4 - i5 / i6);
  b1 || (b2 && b3) || (b4 == b5);

  b1 + i1;           
  // Invalid: mixing bool and int
  b1 - b2;           
  // Invalid: '-' not defined for bools
  i1 && i2;           
  // Invalid: '&&' requires bool operands
  b1 / b2;           
  // Invalid: '/' not defined for bools
  i1 < b1;           
  // Invalid: incompatible types
  b1 >= i1;          
  // Invalid: incompatible types
  (b1 + b2) || b3;   
  // Invalid: '+' not defined for bools
  (i1 == b1) && b2;  
  // Invalid: incompatible types in '=='
  i1 || i2;          
  // Invalid: '||' requires bool operands
  (b1 && i1) == b2;  
  // Invalid: mixing bool and int
  ((i1 + b1) == b2); 
}
