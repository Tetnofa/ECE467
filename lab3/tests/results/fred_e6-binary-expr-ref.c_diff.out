0a1,476
> Program [useIO=0](1,0)
> 	Function[isProto=0]	Int(1,0)
> 		Identifier[name:main](1,4)
> 		Scope(1,11)
> 			Scalar Declaration			Bool(2,2)
> 				Identifier[name:b1](2,7)
> 			Scalar Declaration			Bool(3,2)
> 				Identifier[name:b2](3,7)
> 			Scalar Declaration			Bool(4,2)
> 				Identifier[name:b3](4,7)
> 			Scalar Declaration			Bool(5,2)
> 				Identifier[name:b4](5,7)
> 			Scalar Declaration			Bool(6,2)
> 				Identifier[name:b5](6,7)
> 			Scalar Declaration			Int(7,2)
> 				Identifier[name:i1](7,6)
> 			Scalar Declaration			Int(8,2)
> 				Identifier[name:i2](8,6)
> 			Scalar Declaration			Int(9,2)
> 				Identifier[name:i3](9,6)
> 			Scalar Declaration			Int(10,2)
> 				Identifier[name:i4](10,6)
> 			Scalar Declaration			Int(11,2)
> 				Identifier[name:i5](11,6)
> 			Scalar Declaration			Int(12,2)
> 				Identifier[name:i6](12,6)
> 			Expression Statement(14,2)
> 				Bool Expression(14,2)
> 					Binary Expression[opcode:&&](14,2)
> 						Int Expression(14,2)
> 							Reference(14,2)
> 								Identifier[name:b1](14,2)
> 						Int Expression(14,7)
> 							Reference(14,7)
> 								Identifier[name:b2](14,7)
> 			Expression Statement(15,2)
> 				Bool Expression(15,2)
> 					Binary Expression[opcode:||](15,2)
> 						Int Expression(15,2)
> 							Reference(15,2)
> 								Identifier[name:b1](15,2)
> 						Int Expression(15,8)
> 							Reference(15,8)
> 								Identifier[name:b2](15,8)
> 			Expression Statement(16,2)
> 				Int Expression(16,2)
> 					Binary Expression[opcode:+](16,2)
> 						Int Expression(16,2)
> 							Reference(16,2)
> 								Identifier[name:i1](16,2)
> 						Int Expression(16,7)
> 							Reference(16,7)
> 								Identifier[name:i2](16,7)
> 			Expression Statement(17,2)
> 				Int Expression(17,2)
> 					Binary Expression[opcode:-](17,2)
> 						Int Expression(17,2)
> 							Reference(17,2)
> 								Identifier[name:i1](17,2)
> 						Int Expression(17,7)
> 							Reference(17,7)
> 								Identifier[name:i2](17,7)
> 			Expression Statement(18,2)
> 				Int Expression(18,2)
> 					Binary Expression[opcode:*](18,2)
> 						Int Expression(18,2)
> 							Reference(18,2)
> 								Identifier[name:i1](18,2)
> 						Int Expression(18,6)
> 							Reference(18,6)
> 								Identifier[name:i2](18,6)
> 			Expression Statement(19,2)
> 				Int Expression(19,2)
> 					Binary Expression[opcode:/](19,2)
> 						Int Expression(19,2)
> 							Reference(19,2)
> 								Identifier[name:i1](19,2)
> 						Int Expression(19,7)
> 							Reference(19,7)
> 								Identifier[name:i2](19,7)
> 			Expression Statement(20,2)
> 				Bool Expression(20,2)
> 					Binary Expression[opcode:==](20,2)
> 						Int Expression(20,2)
> 							Reference(20,2)
> 								Identifier[name:i1](20,2)
> 						Int Expression(20,8)
> 							Reference(20,8)
> 								Identifier[name:i2](20,8)
> 			Expression Statement(21,2)
> 				Bool Expression(21,2)
> 					Binary Expression[opcode:!=](21,2)
> 						Int Expression(21,2)
> 							Reference(21,2)
> 								Identifier[name:i1](21,2)
> 						Int Expression(21,8)
> 							Reference(21,8)
> 								Identifier[name:i2](21,8)
> 			Expression Statement(22,2)
> 				Bool Expression(22,2)
> 					Binary Expression[opcode:==](22,2)
> 						Int Expression(22,2)
> 							Reference(22,2)
> 								Identifier[name:b1](22,2)
> 						Int Expression(22,8)
> 							Reference(22,8)
> 								Identifier[name:b2](22,8)
> 			Expression Statement(23,2)
> 				Bool Expression(23,2)
> 					Binary Expression[opcode:!=](23,2)
> 						Int Expression(23,2)
> 							Reference(23,2)
> 								Identifier[name:b1](23,2)
> 						Int Expression(23,8)
> 							Reference(23,8)
> 								Identifier[name:b2](23,8)
> 			Expression Statement(24,2)
> 				Bool Expression(24,2)
> 					Binary Expression[opcode:<](24,2)
> 						Int Expression(24,2)
> 							Reference(24,2)
> 								Identifier[name:i1](24,2)
> 						Int Expression(24,7)
> 							Reference(24,7)
> 								Identifier[name:i2](24,7)
> 			Expression Statement(25,2)
> 				Bool Expression(25,2)
> 					Binary Expression[opcode:<=](25,2)
> 						Int Expression(25,2)
> 							Reference(25,2)
> 								Identifier[name:i1](25,2)
> 						Int Expression(25,8)
> 							Reference(25,8)
> 								Identifier[name:i2](25,8)
> 			Expression Statement(26,2)
> 				Bool Expression(26,2)
> 					Binary Expression[opcode:>](26,2)
> 						Int Expression(26,2)
> 							Reference(26,2)
> 								Identifier[name:i1](26,2)
> 						Int Expression(26,7)
> 							Reference(26,7)
> 								Identifier[name:i2](26,7)
> 			Expression Statement(27,2)
> 				Bool Expression(27,2)
> 					Binary Expression[opcode:>=](27,2)
> 						Int Expression(27,2)
> 							Reference(27,2)
> 								Identifier[name:i1](27,2)
> 						Int Expression(27,8)
> 							Reference(27,8)
> 								Identifier[name:i2](27,8)
> 			Expression Statement(28,2)
> 				Bool Expression(28,2)
> 					Binary Expression[opcode:||](28,2)
> 						Bool Expression(28,3)
> 							Binary Expression[opcode:&&](28,3)
> 								Int Expression(28,3)
> 									Reference(28,3)
> 										Identifier[name:b1](28,3)
> 								Int Expression(28,9)
> 									Reference(28,9)
> 										Identifier[name:b2](28,9)
> 						Int Expression(28,16)
> 							Reference(28,16)
> 								Identifier[name:b3](28,16)
> 			Expression Statement(29,2)
> 				Int Expression(29,2)
> 					Binary Expression[opcode:-](29,2)
> 						Int Expression(29,3)
> 							Binary Expression[opcode:+](29,3)
> 								Int Expression(29,3)
> 									Reference(29,3)
> 										Identifier[name:i1](29,3)
> 								Int Expression(29,8)
> 									Reference(29,8)
> 										Identifier[name:i2](29,8)
> 						Int Expression(29,14)
> 							Reference(29,14)
> 								Identifier[name:i3](29,14)
> 			Expression Statement(30,2)
> 				Int Expression(30,2)
> 					Binary Expression[opcode:/](30,2)
> 						Int Expression(30,3)
> 							Binary Expression[opcode:*](30,3)
> 								Int Expression(30,3)
> 									Reference(30,3)
> 										Identifier[name:i1](30,3)
> 								Int Expression(30,8)
> 									Reference(30,8)
> 										Identifier[name:i2](30,8)
> 						Int Expression(30,15)
> 							Binary Expression[opcode:+](30,15)
> 								Int Expression(30,15)
> 									Reference(30,15)
> 										Identifier[name:i3](30,15)
> 								Int Expression(30,20)
> 									Reference(30,20)
> 										Identifier[name:i4](30,20)
> 			Expression Statement(31,2)
> 				Bool Expression(31,2)
> 					Binary Expression[opcode:&&](31,2)
> 						Bool Expression(31,3)
> 							Binary Expression[opcode:||](31,3)
> 								Int Expression(31,3)
> 									Reference(31,3)
> 										Identifier[name:b1](31,3)
> 								Int Expression(31,9)
> 									Reference(31,9)
> 										Identifier[name:b2](31,9)
> 						Bool Expression(31,17)
> 							Binary Expression[opcode:||](31,17)
> 								Int Expression(31,17)
> 									Reference(31,17)
> 										Identifier[name:b3](31,17)
> 								Int Expression(31,23)
> 									Reference(31,23)
> 										Identifier[name:b4](31,23)
> 			Expression Statement(32,2)
> 				Bool Expression(32,2)
> 					Binary Expression[opcode:<](32,2)
> 						Int Expression(32,3)
> 							Binary Expression[opcode:-](32,3)
> 								Int Expression(32,4)
> 									Binary Expression[opcode:*](32,4)
> 										Int Expression(32,4)
> 											Reference(32,4)
> 												Identifier[name:i1](32,4)
> 										Int Expression(32,9)
> 											Reference(32,9)
> 												Identifier[name:i2](32,9)
> 								Int Expression(32,15)
> 									Reference(32,15)
> 										Identifier[name:i3](32,15)
> 						Int Expression(32,21)
> 							Reference(32,21)
> 								Identifier[name:i4](32,21)
> 			Expression Statement(33,2)
> 				Bool Expression(33,2)
> 					Binary Expression[opcode:==](33,2)
> 						Int Expression(33,3)
> 							Binary Expression[opcode:+](33,3)
> 								Int Expression(33,3)
> 									Reference(33,3)
> 										Identifier[name:i1](33,3)
> 								Int Expression(33,9)
> 									Binary Expression[opcode:*](33,9)
> 										Int Expression(33,9)
> 											Reference(33,9)
> 												Identifier[name:i2](33,9)
> 										Int Expression(33,14)
> 											Reference(33,14)
> 												Identifier[name:i3](33,14)
> 						Int Expression(33,22)
> 							Reference(33,22)
> 								Identifier[name:i4](33,22)
> 			Expression Statement(34,2)
> 				Bool Expression(34,2)
> 					Binary Expression[opcode:==](34,2)
> 						Bool Expression(34,3)
> 							Binary Expression[opcode:&&](34,3)
> 								Int Expression(34,3)
> 									Reference(34,3)
> 										Identifier[name:b1](34,3)
> 								Bool Expression(34,10)
> 									Binary Expression[opcode:||](34,10)
> 										Int Expression(34,10)
> 											Reference(34,10)
> 												Identifier[name:b2](34,10)
> 										Int Expression(34,16)
> 											Reference(34,16)
> 												Identifier[name:b3](34,16)
> 						Bool Expression(34,25)
> 							Binary Expression[opcode:&&](34,25)
> 								Int Expression(34,25)
> 									Reference(34,25)
> 										Identifier[name:b4](34,25)
> 								Int Expression(34,31)
> 									Reference(34,31)
> 										Identifier[name:b5](34,31)
> 			Expression Statement(35,2)
> 				Int Expression(35,2)
> 					Binary Expression[opcode:-](35,2)
> 						Int Expression(35,2)
> 							Binary Expression[opcode:+](35,2)
> 								Int Expression(35,2)
> 									Reference(35,2)
> 										Identifier[name:i1](35,2)
> 								Int Expression(35,7)
> 									Binary Expression[opcode:*](35,7)
> 										Int Expression(35,7)
> 											Reference(35,7)
> 												Identifier[name:i2](35,7)
> 										Int Expression(35,11)
> 											Reference(35,11)
> 												Identifier[name:i3](35,11)
> 						Int Expression(35,16)
> 							Binary Expression[opcode:/](35,16)
> 								Int Expression(35,16)
> 									Reference(35,16)
> 										Identifier[name:i4](35,16)
> 								Int Expression(35,21)
> 									Reference(35,21)
> 										Identifier[name:i5](35,21)
> 			Expression Statement(36,2)
> 				Bool Expression(36,2)
> 					Binary Expression[opcode:<=](36,2)
> 						Int Expression(36,3)
> 							Binary Expression[opcode:+](36,3)
> 								Int Expression(36,3)
> 									Reference(36,3)
> 										Identifier[name:i1](36,3)
> 								Int Expression(36,8)
> 									Binary Expression[opcode:*](36,8)
> 										Int Expression(36,8)
> 											Reference(36,8)
> 												Identifier[name:i2](36,8)
> 										Int Expression(36,13)
> 											Reference(36,13)
> 												Identifier[name:i3](36,13)
> 						Int Expression(36,21)
> 							Binary Expression[opcode:-](36,21)
> 								Int Expression(36,21)
> 									Reference(36,21)
> 										Identifier[name:i4](36,21)
> 								Int Expression(36,26)
> 									Binary Expression[opcode:/](36,26)
> 										Int Expression(36,26)
> 											Reference(36,26)
> 												Identifier[name:i5](36,26)
> 										Int Expression(36,31)
> 											Reference(36,31)
> 												Identifier[name:i6](36,31)
> 			Expression Statement(37,2)
> 				Bool Expression(37,2)
> 					Binary Expression[opcode:||](37,2)
> 						Bool Expression(37,2)
> 							Binary Expression[opcode:||](37,2)
> 								Int Expression(37,2)
> 									Reference(37,2)
> 										Identifier[name:b1](37,2)
> 								Bool Expression(37,9)
> 									Binary Expression[opcode:&&](37,9)
> 										Int Expression(37,9)
> 											Reference(37,9)
> 												Identifier[name:b2](37,9)
> 										Int Expression(37,15)
> 											Reference(37,15)
> 												Identifier[name:b3](37,15)
> 						Bool Expression(37,23)
> 							Binary Expression[opcode:==](37,23)
> 								Int Expression(37,23)
> 									Reference(37,23)
> 										Identifier[name:b4](37,23)
> 								Int Expression(37,29)
> 									Reference(37,29)
> 										Identifier[name:b5](37,29)
> 			Expression Statement(39,2)
> 				Int Expression(39,2)
> 					Binary Expression[opcode:+](39,2)
> 						Int Expression(39,2)
> 							Reference(39,2)
> 								Identifier[name:b1](39,2)
> 						Int Expression(39,7)
> 							Reference(39,7)
> 								Identifier[name:i1](39,7)
> 			Expression Statement(41,2)
> 				Int Expression(41,2)
> 					Binary Expression[opcode:-](41,2)
> 						Int Expression(41,2)
> 							Reference(41,2)
> 								Identifier[name:b1](41,2)
> 						Int Expression(41,7)
> 							Reference(41,7)
> 								Identifier[name:b2](41,7)
> 			Expression Statement(43,2)
> 				Bool Expression(43,2)
> 					Binary Expression[opcode:&&](43,2)
> 						Int Expression(43,2)
> 							Reference(43,2)
> 								Identifier[name:i1](43,2)
> 						Int Expression(43,8)
> 							Reference(43,8)
> 								Identifier[name:i2](43,8)
> 			Expression Statement(45,2)
> 				Int Expression(45,2)
> 					Binary Expression[opcode:/](45,2)
> 						Int Expression(45,2)
> 							Reference(45,2)
> 								Identifier[name:b1](45,2)
> 						Int Expression(45,7)
> 							Reference(45,7)
> 								Identifier[name:b2](45,7)
> 			Expression Statement(47,2)
> 				Bool Expression(47,2)
> 					Binary Expression[opcode:<](47,2)
> 						Int Expression(47,2)
> 							Reference(47,2)
> 								Identifier[name:i1](47,2)
> 						Int Expression(47,7)
> 							Reference(47,7)
> 								Identifier[name:b1](47,7)
> 			Expression Statement(49,2)
> 				Bool Expression(49,2)
> 					Binary Expression[opcode:>=](49,2)
> 						Int Expression(49,2)
> 							Reference(49,2)
> 								Identifier[name:b1](49,2)
> 						Int Expression(49,8)
> 							Reference(49,8)
> 								Identifier[name:i1](49,8)
> 			Expression Statement(51,2)
> 				Bool Expression(51,2)
> 					Binary Expression[opcode:||](51,2)
> 						Int Expression(51,3)
> 							Binary Expression[opcode:+](51,3)
> 								Int Expression(51,3)
> 									Reference(51,3)
> 										Identifier[name:b1](51,3)
> 								Int Expression(51,8)
> 									Reference(51,8)
> 										Identifier[name:b2](51,8)
> 						Int Expression(51,15)
> 							Reference(51,15)
> 								Identifier[name:b3](51,15)
> 			Expression Statement(53,2)
> 				Bool Expression(53,2)
> 					Binary Expression[opcode:&&](53,2)
> 						Bool Expression(53,3)
> 							Binary Expression[opcode:==](53,3)
> 								Int Expression(53,3)
> 									Reference(53,3)
> 										Identifier[name:i1](53,3)
> 								Int Expression(53,9)
> 									Reference(53,9)
> 										Identifier[name:b1](53,9)
> 						Int Expression(53,16)
> 							Reference(53,16)
> 								Identifier[name:b2](53,16)
> 			Expression Statement(55,2)
> 				Bool Expression(55,2)
> 					Binary Expression[opcode:||](55,2)
> 						Int Expression(55,2)
> 							Reference(55,2)
> 								Identifier[name:i1](55,2)
> 						Int Expression(55,8)
> 							Reference(55,8)
> 								Identifier[name:i2](55,8)
> 			Expression Statement(57,2)
> 				Bool Expression(57,2)
> 					Binary Expression[opcode:==](57,2)
> 						Bool Expression(57,3)
> 							Binary Expression[opcode:&&](57,3)
> 								Int Expression(57,3)
> 									Reference(57,3)
> 										Identifier[name:b1](57,3)
> 								Int Expression(57,9)
> 									Reference(57,9)
> 										Identifier[name:i1](57,9)
> 						Int Expression(57,16)
> 							Reference(57,16)
> 								Identifier[name:b2](57,16)
> 			Expression Statement(59,2)
> 				Bool Expression(59,3)
> 					Binary Expression[opcode:==](59,3)
> 						Int Expression(59,4)
> 							Binary Expression[opcode:+](59,4)
> 								Int Expression(59,4)
> 									Reference(59,4)
> 										Identifier[name:i1](59,4)
> 								Int Expression(59,9)
> 									Reference(59,9)
> 										Identifier[name:b1](59,9)
> 						Int Expression(59,16)
> 							Reference(59,16)
> 								Identifier[name:b2](59,16)
